pkg/providers/cloudstack/cloudstack.go

```diff
@@ -530,13 +531,14 @@ func AnyImmutableFieldChanged(oldCsdc, newCsdc *v1alpha1.CloudStackDatacenterCon
 	if oldCsmc.Spec.ComputeOffering != newCsmc.Spec.ComputeOffering {
 		return true
 	}
-	if len(oldCsmc.Spec.UserCustomDetails) != len(newCsmc.Spec.UserCustomDetails) {
+	if !reflect.DeepEqual(oldCsmc.Spec.UserCustomDetails, newCsmc.Spec.UserCustomDetails) {
```

Comment:
the general consensus is that we should avoid using reflect.DeepEqual outside of test code. Please revert this change